php/site-utils.php

```diff
@@ -143,27 +143,55 @@ function setup_site_network( $site_name ) {
 
 }
 
+/**
+ * Reloads configuration of ee-nginx-proxy container
+ *
+ * @return bool
+ */
+function reload_proxy_configuration() {
+	return EE::exec( 'docker exec ee-nginx-proxy sh -c "/app/docker-entrypoint.sh /usr/local/bin/docker-gen /app/nginx.tmpl /etc/nginx/conf.d/default.conf; /usr/sbin/nginx -s reload"' );
+}
+
 /**
  * Adds www to non-www redirection to site
  *
- * @param string $site_name Name of the site.
- * @param bool $le          Specifying if letsencrypt is enabled or not.
+ * @param string $site_name name of the site.
+ * @param bool $ssl         enable ssl or not.
+ * @param bool $inherit     inherit cert or not.
  */
-function add_site_redirects( $site_name, $le ) {
+function add_site_redirects( string $site_name, $ssl, $inherit ) {
 
 	$fs               = new Filesystem();
 	$confd_path       = EE_CONF_ROOT . '/nginx/conf.d/';
 	$config_file_path = $confd_path . $site_name . '-redirect.conf';
 	$has_www          = strpos( $site_name, 'www.' ) === 0;
+	$cert_site_name   = $site_name;
+
+	if ( $inherit ) {
+		$cert_site_name = implode( '.', array_slice( explode( '.', $site_name ), 1 ) );
+	}
 
 	if ( $has_www ) {
 		$site_name_without_www = ltrim( $site_name, '.www' );
 		// ee site create www.example.com --le
-		if ( $le ) {
+		if ( $ssl ) {
 			$content = "
 server {
 	listen  80;
+	server_name  $site_name_without_www;
```

Comment:
Can this go in mustache template? It will be easy to read and update config in that.